nearcore/src/lib.rs

```diff
@@ -589,9 +624,27 @@ pub fn recompress_storage(home_dir: &Path, dst_dir: &Path) -> anyhow::Result<()>
         store_update.commit()?;
     }
 
+    // If we’re not keeping ColPartialChunks, update chunk tail to point to
+    // current final block.  If we don’t do that, the gc will try to work its
+    // way from the genesis even though chunks at those heights have been
+    // deleted.
+    if skip_columns.contains(&DBCol::ColPartialChunks) {
+        let tip: Option<near_primitives::block::Tip> =
+            src_store.get_ser(DBCol::ColBlockMisc, near_store::FINAL_HEAD_KEY)?;
+        if let Some(tip) = tip {
+            let chunk_tail: near_primitives::types::BlockHeight = tip.height;
+            info!("Setting chunk tail to {}", chunk_tail);
+            let mut store_update = dst_store.store_update();
+            store_update.set_ser(DBCol::ColBlockMisc, near_store::CHUNK_TAIL_KEY, &chunk_tail)?;
+            store_update.commit()?;
+        } else {
+            info!("Final head not set; leaving chunk tail be");
```

Comment:
what are the scenarios when we're missing the 'final head'?

Reply:
In any legitimate uses of recompress storage I can’t really think of any scenario. Do you think the code should just panic?